internal/scheduler/schedulerobjects/resourcelist.go

```diff
@@ -39,83 +39,85 @@ func V1ResourceListFromResourceList(rl ResourceList) v1.ResourceList {
 	return rv
 }
 
-type QuantityByPriorityAndResourceType map[int32]ResourceList
+type QuantityByTAndResourceType[T comparable] map[T]ResourceList
```

Comment:
I've replaces `QuantityByPriorityAndResourceType` with the more generic `QuantityByTAndResourceType[int32]`, which also allows us to use `QuantityByTAndResourceType[string]`, to track resources by priority class name.